### PR TITLE
Add VPS hosting guide

### PR DESCRIPTION
**Pull request by rojserbest**

### CODE REVIEW

**Comment by rojserbest** [comment]
This should be split into more parts depending on the tool used to manage the bot process, and in the beginning we will have a small grammY-VPS talk and add anchor links to all methods.

**Comment by rojserbest** [comment]
Also, in the end we can have a guide about setting up webhooks with nginx and Let's Encrypt SSL.

File: site/docs/.vuepress/config.ts [hunk 1]
```diff
@@ -323,6 +323,10 @@ export default defineUserConfig<DefaultThemeOptions>({
                     text: "Google Cloud Functions",
                     link: "/hosting/gcf.md",
                   },
+                  {
+                    text: "VPS",
```
Comment (KnorpelSenf) [inline site/docs/.vuepress/config.ts:327]: Should we give this a name that people understand who are not familiar with the abbreviation VPS?

File: site/docs/hosting/vps.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
+# VPS
+
+A virtual private server, mostly known as VPS is a virtual machine running in the cloud with its users having the full control of its filesystem.
```
Comment (KnorpelSenf) [inline site/docs/hosting/vps.md:3]:
```suggestion
A virtual private server, mostly known as VPS, is a virtual machine running in the cloud with its users having the full control of its filesystem.
```

File: site/docs/hosting/vps.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+There are tons of advantages in using a VPS over things like Heroku and microservices to run your bots.
```
Comment (KnorpelSenf) [inline site/docs/hosting/vps.md:4]: This is only partially true. Yes, you have more control, but it also means that you have more responsibility. Heroku manages system updates and security patches automatically without downtime or intervention, which is very hard to do manually. I am uncomfortable with painting a black-and-white picture here.

File: site/docs/hosting/vps.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+In this guide, you'll learn about various methods of running your bot in a VPS, keeping it online 24/7, making it run automatically when your VPS boots and restart on crashes.
+
+## Methods of Hosting
+
+- [systemd](#systemd)
+
+---
+
+# systemd
```
Comment (KnorpelSenf) [inline site/docs/hosting/vps.md:14]: Please check out the deploy preview. This structure does not render well. The link on the side are missing.

File: site/docs/hosting/vps.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+```bash
+systemctl daemon-reload
+```
+
+## Managing the service
+
+### Start
+
+```bash
+systemctl start <service_name>
```
Comment (KnorpelSenf) [inline site/docs/hosting/vps.md:88]: Please be more clear above what the service name is, so there's no doubt about what to type here.

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]

File: site/docs/hosting/vps.md (hunk 5 above)
Comment (rojserbest) [reply site/docs/hosting/vps.md:75]: I guess I have a blockquote telling that. Is there a way to be clearer that I don't know?

Comment (KnorpelSenf) [reply site/docs/hosting/vps.md:75]: No sorry, I misread. I thought you said just “the name of the service”. Mentioning the file name is good.

**Comment by KnorpelSenf** [comment]
Waiting for translations.

**Comment by rojserbest** [comment]
@KnorpelSenf please don't delete the `vps-hosting` branch in case if you merged this.

File: site/docs/hosting/vps.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+### Managing the service
+
+#### Start
+
+```bash
+systemctl start <service_name>
+```
+
+> Replace `<service_name>` with the file name of the service.
```
Comment (WingLim) [inline site/docs/hosting/vps.md:78]: Maybe we should add a example, like this:
```suggestion
> Replace `<service_name>` with the file name of the service.
> Example: `systemctl start bot1`
```

**Review by WingLim — APPROVED** [review]

**Review by WingLim — APPROVED** [review]
🚀

**Pull request merged**